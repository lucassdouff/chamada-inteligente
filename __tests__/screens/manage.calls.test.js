import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContext } from '@react-navigation/native';
import { Location } from 'expo';
import ManageCallsScreen from '../../src/app/screens/teacher/ManageCalls/ManageCalls';

const navContext = {
    isFocused: () => true,
    addListener: jest.fn(() => jest.fn())
}

jest.mock('expo', () => ({
    Location: {
      requestForegroundPermissionsAsync: jest.fn(),
      getCurrentPositionAsync: jest.fn(),
    },
}));

describe('Manage calls screen should render correctly', () => {

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('teacher manage calls renders correctly', async () => {

        Location.requestForegroundPermissionsAsync.mockResolvedValueOnce({ status: 'granted' });
        
        const mockedParams = {
            route: { params: { userClass: {
                id_class: 1,
                name: 'Class 1',
                code: 123456,
                semester: '2021-1',
                id_teacher: 1,
                id_course: 1,
                class_weekdays: [{
                    weekday: 1,
                    start_hour: '08:00:00',
                    end_hour: '10:00:00'
                }]
            }}
            },
            navigation: ''
        };

        await expect(render(
            <NavigationContext.Provider value={navContext}>
                <ManageCallsScreen {...mockedParams} />
            </NavigationContext.Provider>
        )).toBeTruthy();
    });
    
});
