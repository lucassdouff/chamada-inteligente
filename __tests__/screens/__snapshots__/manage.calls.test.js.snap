// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage calls screen should render correctly teacher manage calls renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      {
        "width": "100%",
      },
      {
        "marginTop": 8,
      },
      {
        "height": "100%",
      },
    ]
  }
>
  <View>
    <View>
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "paddingBottom": 4,
              "paddingTop": 4,
            },
          ]
        }
      >
        Chamadas Agendadas
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "center",
            },
            {
              "marginTop": 16,
            },
            {
              "paddingTop": 24,
            },
            [
              {
                "borderBottomColor": "#6b7280",
                "borderLeftColor": "#6b7280",
                "borderRightColor": "#6b7280",
                "borderTopColor": "#6b7280",
              },
              {
                "borderBottomWidth": 0,
                "borderTopWidth": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "width": "100%",
              },
            ]
          }
        >
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  [
                    {
                      "flexDirection": "column",
                    },
                    {
                      "borderBottomWidth": 1,
                      "borderLeftWidth": 1,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                  ]
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "marginBottom": 32,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginBottom": 8,
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "green",
                  },
                ]
              }
            >
              AGENDAR NOVA CHAMADA
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      visible={false}
    />
  </View>
</RCTScrollView>
`;
